lesson-04-memory-management/mm/xxx.c

```diff
@@ -35,20 +146,37 @@ int __init x_init(void)
 {
 	int res;
 
+   pr_info("mem_type is %d\n", mem_type);
 	x_class = class_create(THIS_MODULE, "x-class");
 	if (IS_ERR(x_class))
 		pr_info("bad class create\n");
 	res = class_create_file(x_class, &class_attr_xxx);
```

Comment:
`xxx_store` may already be called at this point,
while `pmem_cache` hasn't yet been created.